### ISSUE TITLE
Fetching complete history and spinning feature in Random Name Picker

### ISSUE THREAD

**Issue by cckole** [Gear61/Random-Name-Picker#31]
COMPONENT

Random name picker is an open source android application that fundamentally help you choose from a list of names at random. This app display names that are literally listed and are chosen at random for all intent and purpose depending on the chosen number of names to be displayed. In the chosen name history, only the chosen name is shown, time and date isn't displayed. Also the random pick is only one way, there isn't any way to spin other than clicking on CHOSE to show the name picked at random which made the app to be limited. Going further, i will give list of some features to enhance the project and i hope it would be put into consideration.

PROPOSAL

This app was built to randomly pick names listed under list name and it has been a very good measure taken to make the app a good one, However, i will like to add my points to further enhance it in other to make users have more fun using Random Name Picker.

DATE AND TIME
When name is picked at random from the app, the picked name is stored somewhere in the chosen name history which can later be accessed when such information is needed but no information is complete without date and time of occurrence which is missing in the history. The date and time each name is randomly picked should be included in the history for reference purpose .since the names are already recorded, the date and time can as well be placed at the front of each name in the "chosen name history. This will make the date and time traceable when needed.

SPIN THE ROULETTE

Since the app can only pick numbers at random by clicking on the "CHOSE" button below the listed names. It will be fun to have an alternative way of choosing names at random as the current design doesn't welcome some approaches .
To implement the spinning feature, the page on the app where names are picked at random will have a part as shown in the mock up that will be clicked to display a roulette in form of a pie chat. After names to be picked to random has been listed,the spin button will be clicked which will result in the display of a page showing the names in a circle (Roulette) . user can then click on spin below to select which ever name is pointed at by the pointer the moment the spinning circle stops.
This spinning feature can go a long way in making users catch fun with the app. The benefit below will shed more light and create some interesting effect.

MOCKS/UP

The images below shows the look of Random Name Picker with listed names
5678.png  1234.png
comparison between the present history page and the proposed one
name history.png  name history 1234.png

The date and time will make the event track able any day any time.
From the image below, clicking the SPIN will open another page where a ROULETTE can be accessed to spin
hjhjhjhjhjhj.png
party games.png  who to answer question.png
from the above images, the spinning feature can be used for fun by picking someone randomly and spinning the roulette to chose a task for the chosen person to perform...

BENEFIT
Having stated the features that might enhance the app, I believe it will go a long way in making it more unique.
The date and time will allow the user to get the accurate history of activities done on the app. For instance, if a raffle draw is organized and the app is used as the tool to pick names at random, it would be easier and stress free to get the information of the event as touching name of winners, Time and the date the raffle took place after the event is long gone. Sometimes, draws are drawn at interval or even days in between, so if date and time are not present, it would be difficult to get the accurate history of the activities. Though there might be a way to go about it, it would be inappropriate to get the date and time elsewhere which in some cases might not even be correct if proper arrangement was not made in the first place.
Really this might look simple but it will surely be of benefit.

The spinning feature will allow users to do more things with the app. apart from picking names by spinning, it can be used in birthday parties, home, schools and even get together events. Series of things can be done with the spinning feature.
As Random Picker normally randomly picks item from pre-defined lists setup on device. The spinning feature will integrate the process by making it more fun and exciting.
It can be used to groove a party!
People can watch in suspense as their name stays in the running – or is eliminated – until the final random name is left! Watching as the roulette spins will create a kind of excitement as people watch and wait to see who gets the pointer.
Choosing the random name and activities is part of the fun – a great attention from viewers

In conclusion
The Bar chart and pie chart are different but they work with the same list in arithmetic. So also the spinning roulette and current design of random name picker, they can virtually work on predefined list set up by user but the fun, excitement, and the presentation of the end product is incomparable.

**Comment by Gear61** [comment]
I feel like the wheel spinning is out of scope for this app. I want this app to remain lean and focused; there are already other apps that do that. Also, your mocks don't show up.
